labrad/decorators.py

```diff
@@ -21,202 +21,140 @@
 
 from __future__ import absolute_import
 
-from functools import wraps
-from inspect import getargspec
+import functools
+import inspect
+import itertools
 
-from twisted.internet.defer import inlineCallbacks
+import twisted.internet.defer as defer
 
 from labrad import types as T, util
 
-def _isGenerator(f):
-    """Check to see whether f is a generator.
-
-    See the documentation on code objects at:
-    http://docs.python.org/ref/types.html
-    """
-    return bool(f.func_code.co_flags & 0x20)
-
 def _product(lists):
     """Return the cartesian product of a list of lists."""
     if not len(lists): return [[]]
     return [[h] + t for h in lists[0] for t in _product(lists[1:])]
 
-class Setting(object):
-    def __init__(self, func):
-        self.func = func
+def setting(lr_ID, lr_name=None, returns=[], unflatten=True, **params):
+    def decorator(func):
+        return Setting(func, lr_ID, lr_name, returns, unflatten, **params)
+    return decorator
 
-    def getRegistrationInfo(self):
-        return (long(self.ID), self.name, self.description,
-                self.accepts, self.returns, self.notes)
+class Setting(object):
+    """
+    Represent a single setting within a Server object.
 
-    def handleRequest(self, c, data):
-        return self.func(c, data)
+    args:
```

Comment:
Please follow the formatting conventions in the [google python style guide](https://google.github.io/styleguide/pyguide.html?showone=Comments#Comments) and include types:
```python
Args:
    func (callable): Function to implement the setting.
    ...
```